File: 6/Refactoring/src/controller/SearchController.java
package controller;

import model.Shop;
import model.User;
import model.Search;

import java.util.HashMap;

public class SearchController {
    private Shop shopForSearching;
    public Shop getShopForSearching() {
        return shopForSearching;
    }
    public void setShopForSearching(Shop shopForSearcing) {
        this.shopForSearching = shopForSearcing;
    }

    private User initialUSer;
    public User getInitialUSer() {
        return initialUSer;
    }
    public void setInitialUSer(User initialUSer) {
        this.initialUSer = initialUSer;
    }

    public SearchController(Shop shopForSearching, User initialUSer) {
        this.shopForSearching = shopForSearching;
        this.initialUSer = initialUSer;
    }
    // Запрашиваем у пользователя критерии, по которым он хочет сделать фильтр и печатаем их.
    public void newSearch (){
        this.initialUSer.setRequestedCriteria();
        this.initialUSer.printCriteria();

        // Запрашиваем у пользователя конкретные характеристики для фильтрации
        this.initialUSer.setRequestedChars();
        HashMap<String, String> filterMap = this.initialUSer.getRequestedChars();

        // Переформатируем требования пользователя в запрос для магазина, выводим пользователю его запрос
        this.shopForSearching.setRequest(filterMap);
        this.shopForSearching.getRequest().printRequest();

        // Создадим фильтр по поступившему запросу:
        Search newUserFilter = new Search(this.shopForSearching.getRequest());
        this.initialUSer.setFilter(newUserFilter);
    }
}

File: 7/controller/search_controller.py
from model.Search_model import Search


class Search_Controller:
    __shop_for_search = None
    __initial_user = None

    def __init__(self, shop, user):
        self.__shop_for_search = shop
        self.__initial_user = user

    def get_shop(self):
        return self.__shop_for_search

    def set_shop(self, shop):
        self.__shop_for_search = shop

    def get_user(self):
        return self.__initial_user

    def set_user(self, user):
        self.__initial_user = user

    def new_search(self):
        self.__initial_user.set_requested_criteria()
        self.__initial_user.print_criteria()

        self.__initial_user.set_requested_chars()
        filter_dict = self.__initial_user.get_requested_chars()

        self.__shop_for_search.set_request(filter_dict)
        print(self.__shop_for_search.get_request())

        user_filter = Search(self.__shop_for_search.get_request())
        self.__initial_user.set_filter(user_filter)
